### PR TITLE
Task8: Implemented CountUniqueWords.java

### PR DESCRIPTION
**Pull request by vminin-solvd**
Used FileUtils and StringUtils to count unique (lowercased) words, static method is used in Main.java.

### CODE REVIEW

File: src/main/java/com/solvd/zoo/wordcounter/CountUniqueWords.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                }
+            }
+            int uniqueWords = wordCountMap.size();
+            String newFileContent = fileContent + " " + uniqueWords;
+            FileUtils.writeStringToFile(file, newFileContent, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            LOGGER.error("Error occurred while processing the file", e);
+        }
+    }
+    private static String getResourcePath(String resourceName) {
```
Comment (ankochergin) [inline src/main/java/com/solvd/zoo/wordcounter/CountUniqueWords.java]:
```suggestion
    }
    
    private static String getResourcePath(String resourceName) {
```

File: src/main/java/com/solvd/zoo/wordcounter/CountUniqueWords.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+public class CountUniqueWords {
+
+    private static final Logger LOGGER = LogManager.getLogger(Visitor.class);
+
+    public static void countUniqueWord() {
+
+        try {
+            File file = new File(getResourcePath("FamousPhrases.txt"));
+            String fileContent = FileUtils.readFileToString(file, StandardCharsets.UTF_8);
```
Comment (ankochergin) [inline src/main/java/com/solvd/zoo/wordcounter/CountUniqueWords.java]:
```suggestion
    public static void countUniqueWord(String fileName) {
        try {
            File file = new File(getResourcePath(fileName);
            String fileContent = FileUtils.readFileToString(file, StandardCharsets.UTF_8);
```

File: src/main/java/com/solvd/zoo/Main.java [hunk 3]
```diff
@@ -13,16 +13,18 @@
...
 import static com.solvd.zoo.Zoo.welcomeVisitor;
+import static com.solvd.zoo.wordcounter.CountUniqueWords.countUniqueWord;
 
 public class Main {
 
     private static final Logger LOGGER = LogManager.getLogger(Main.class);
 
-    public static void main(String[] args) throws InvalidNameException, LocationException, ExpiredTicketException, FeedAnimalException {
-
+    public static void main(String[] args) throws InvalidNameException, LocationException, ExpiredTicketException, FeedAnimalException, IOException {
```
Comment (ankochergin) [inline src/main/java/com/solvd/zoo/Main.java]: please don't throw exceptions from main, handle them all using try-catch

Comment (vminin-solvd) [reply src/main/java/com/solvd/zoo/Main.java]: Noted.

File: src/main/java/com/solvd/zoo/wordcounter/CountUniqueWords.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CountUniqueWords {
+
+    public static void countUniqueWord(String fileName) throws IOException{
+
+            File file = new File(getResourcePath(fileName));
+            String fileContent = FileUtils.readFileToString(file, StandardCharsets.UTF_8);
```
Comment (ankochergin) [inline src/main/java/com/solvd/zoo/wordcounter/CountUniqueWords.java]:
```suggestion
    public static void countUniqueWord(String fileName) throws IOException{
            File file = new File(getResourcePath(fileName));
            String fileContent = FileUtils.readFileToString(file, StandardCharsets.UTF_8);
```

**Pull request merged**